modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -1651,22 +1659,69 @@ bool LMDBBackend::createDomain(const DNSName& domain, const DomainInfo::DomainKi
   return true;
 }
 
+void LMDBBackend::getAllDomainsFiltered(vector<DomainInfo>* domains, const std::function<bool(DomainInfo&)>& allow)
+{
+  if (d_handle_dups) {
+    auto txn = d_tdomains->getROTransaction();
+
+    map<DNSName, DomainInfo> zonemap;
+    set<DNSName> dups;
+
+    for (auto iter = txn.begin(); iter != txn.end(); ++iter) {
+      DomainInfo di = *iter;
+      di.id = iter.getID();
+      di.backend = this;
+
+      if (zonemap.count(di.zone) == 1) {
+        dups.insert(di.zone);
+      }
+      else {
+        zonemap[di.zone] = di;
+      }
+    }
+
+    for (const auto& zone : dups) {
+      DomainInfo di;
+
+      if (!(di.id = txn.get<0>(zone, di))) {
```

Comment:
I'm always wary of setting a value in a condition, I find it easy to misunderstand.

Reply:
Do I correctly understand that this works because `get` will return the most recent entry? If so this might need a comment.

Reply:
the least recent - I'll add a comment!

Reply:
also moved get outside of if